src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.LeadingZeroCount.cs

```diff
@@ -29,18 +29,52 @@ public static void LeadingZeroCount<T>(ReadOnlySpan<T> x, Span<T> destination)
         internal readonly unsafe struct LeadingZeroCountOperator<T> : IUnaryOperator<T, T> where T : IBinaryInteger<T>
         {
             public static bool Vectorizable =>
-                (Avx512CD.VL.IsSupported && (sizeof(T) == 4 || sizeof(T) == 8)) ||
+                (Avx512CD.VL.IsSupported && (sizeof(T) == 2 || sizeof(T) == 4 || sizeof(T) == 8)) ||
+                (Avx512BW.IsSupported && Avx512Vbmi.IsSupported && Avx512Vbmi.VL.IsSupported && sizeof(T) == 1) ||
                 (AdvSimd.IsSupported && (sizeof(T) == 1 || sizeof(T) == 2 || sizeof(T) == 4));
 
             public static T Invoke(T x) => T.LeadingZeroCount(x);
 
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
             public static Vector128<T> Invoke(Vector128<T> x)
             {
+                if (Avx512Vbmi.VL.IsSupported && sizeof(T) == 1)
+                {
+                    Vector128<byte> lookupVectorLow = Vector128.Create((byte)8, 7, 6, 6, 5, 5, 5, 5, 4, 4, 4, 4, 4, 4, 4, 4);
+                    Vector128<byte> lookupVectorHigh = Vector128.Create((byte)3, 2, 2, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0);
+                    Vector128<byte> nibbleMask = Vector128.Create<byte>(0xF);
+                    Vector128<byte> permuteMask = Vector128.Create<byte>(0x80);
+                    Vector128<byte> lowNibble = x.AsByte() & nibbleMask;
+                    Vector128<byte> highNibble = Sse2.ShiftRightLogical(x.AsInt32(), 4).AsByte() & nibbleMask;
+                    Vector128<byte> nibbleSelectMask = Sse2.CompareEqual(highNibble, Vector128<byte>.Zero);
+                    Vector128<byte> indexVector = Sse41.BlendVariable(highNibble, lowNibble, nibbleSelectMask) +
+                        (~nibbleSelectMask & nibbleMask);
+                    indexVector |= ~nibbleSelectMask & permuteMask;
+                    return Avx512Vbmi.VL.PermuteVar16x8x2(lookupVectorLow, indexVector, lookupVectorHigh).As<byte, T>();
+                }
+
                 if (Avx512CD.VL.IsSupported)
                 {
-                    if (sizeof(T) == 4) return Avx512CD.VL.LeadingZeroCount(x.AsUInt32()).As<uint, T>();
-                    if (sizeof(T) == 8) return Avx512CD.VL.LeadingZeroCount(x.AsUInt64()).As<ulong, T>();
+                    if (sizeof(T) == 2)
+                    {
+                        Vector128<uint> lowHalf = Vector128.Create((uint)0x0000FFFF);
+                        Vector128<uint> x_bot16 = Sse2.Or(Sse2.ShiftLeftLogical(x.AsUInt32(), 16), lowHalf);
+                        Vector128<uint> x_top16 = Sse2.Or(x.AsUInt32(), lowHalf);
+                        Vector128<uint> lz_bot16 = Avx512CD.VL.LeadingZeroCount(x_bot16);
+                        Vector128<uint> lz_top16 = Avx512CD.VL.LeadingZeroCount(x_top16);
+                        Vector128<uint> lz_top16_shift = Sse2.ShiftLeftLogical(lz_top16, 16);
+                        return Sse2.Or(lz_bot16, lz_top16_shift).AsUInt16().As<ushort, T>();
```

Comment:
Is this cheaper than:
```csharp
Vector256<int> x32 = Avx2.ConvertToVector256Int32(x.AsUInt16());
Vector256<int> lz = Avx512CD.VL.LeadingZeroCount(x32);
return Avx512F.VL.ConvertToVector128UInt16(lz);
```

Reply:
This comment (and the related ones below) are still pending a response.

The current pattern used looks significantly more expensive (in size, instruction count, and micro-ops) than the more naive method of `widen + lzcnt + narrow + subtract`

Reply:
One more thing to highlight in the above codegen is that the current PR produces 9 instructions for each invocation of LeadingZeroCount. Using the widen+narrow approach produces 11 instructions.

Original loop is unrolled, I'm copying/pasting the codegen of just one invocation of LeadingZeroCount:

**Current PR** -- 9 total instructions
```
vmovups zmm3, zmmword ptr [rdx] 
vmovaps zmm4, zmm1 
vpslld zmm5, zmm3, 16 
vpord zmm5, zmm5, zmm4 
vplzcntd zmm5, zmm5 
vpord zmm3, zmm3, zmm4 
vplzcntd zmm3, zmm3 
vpslld zmm3, zmm3, 16 
vpord zmm3, zmm3, zmm5
```

**Widen+Narrow** -- 11 total instructions
```
vmovups zmm0, zmmword ptr [rdx] 
vextracti32x8 ymm1, zmm0, 1 
vpmovzxwd zmm2, zmm1 
vplzcntd zmm3, zmm2 
vpmovzxwd zmm4, zmm0 
vplzcntd zmm5, zmm4 
vpmovdw zmm5, zmm16 
vpmovdw zmm3, zmm17 
vinsertf64x4 zmm18, zmm18, ymm16, 0 
vinsertf64x4 zmm18, zmm18, ymm17, 1 
vpsubw zmm0, zmm18, zmm27
```

I'm still proposing the current PR as it produces faster and smaller codegen.